packages/block-library/src/latest-posts/edit.native.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * External dependencies
+ */
+import { TouchableWithoutFeedback, View, Text } from 'react-native';
+import { isEmpty } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { compose, withPreferredColorScheme } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+import { coreBlocks } from '@wordpress/block-library';
+import { __ } from '@wordpress/i18n';
+import { postList as icon } from '@wordpress/icons';
+import { InspectorControls } from '@wordpress/block-editor';
+import { fetchRequest } from 'react-native-gutenberg-bridge';
+import {
+	Icon,
+	PanelBody,
+	ToggleControl,
+	RangeControl,
+	QueryControls,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import { MIN_EXCERPT_LENGTH, MAX_EXCERPT_LENGTH } from './constants';
+
+class LatestPostsEdit extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			categoriesList: [],
+		};
+		this.onSetDisplayPostContent = this.onSetDisplayPostContent.bind(
+			this
+		);
+		this.onSetDisplayPostContentRadio = this.onSetDisplayPostContentRadio.bind(
+			this
+		);
+		this.onSetExcerptLength = this.onSetExcerptLength.bind( this );
+		this.onSetDisplayPostDate = this.onSetDisplayPostDate.bind( this );
+		this.onSetOrder = this.onSetOrder.bind( this );
+		this.onSetOrderBy = this.onSetOrderBy.bind( this );
+		this.onSetPostsToShow = this.onSetPostsToShow.bind( this );
+		this.onSetCategories = this.onSetCategories.bind( this );
+		this.getInspectorControls = this.getInspectorControls.bind( this );
+	}
+
+	componentDidMount() {
+		this.isStillMounted = true;
+		this.fetchRequest = fetchRequest( '/wp/v2/categories' )
+			.then( ( categoriesList ) => {
+				if ( this.isStillMounted ) {
+					let parsedCategoriesList = categoriesList;
+
+					// TODO: remove this check after `fetchRequest` types are made consist across platforms
+					// (see: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1961)
+					if ( typeof categoriesList === 'string' ) {
+						parsedCategoriesList = JSON.parse( categoriesList );
+					}
+
+					if ( isEmpty( parsedCategoriesList ) ) {
```

Comment:
The demo app was returning an object on its failure to fetch categories. I tried to add `typeof categoriesList === 'array'` here but got a lint error. So instead I'm checking to see if it's empty which seems to handle the weird object case.